web/modules/custom/joinup_news/src/Controller/NewsController.php

```diff
@@ -64,14 +64,25 @@ public function add(RdfInterface $rdf_entity) {
   /**
    * Handles access to the news add form through rdf entity pages.
    *
+   * Access is granted to moderators and group members that have the permission
+   * to create news articles inside of their group, which in practice means this
+   * is granted to collection and solution facilitators.
+   *
    * @param \Drupal\rdf_entity\RdfInterface $rdf_entity
    *   The RDF entity for which the news entity is created.
    *
    * @return \Drupal\Core\Access\AccessResult
    *   The access result object.
    */
   public function createNewsAccess(RdfInterface $rdf_entity) {
-    return $this->ogAccess->userAccessGroupContentEntityOperations('create', $rdf_entity, $this->createNewsEntity($rdf_entity), $this->currentUser());
+    $user = $this->currentUser();
+    // Grant access if the user is a moderator.
+    if (in_array('moderator', $user->getRoles())) {
```

Comment:
I know the ticket is about restoring the create permissions to the moderator but is this the proper way to go? I mean hardcoding the moderator role in the permission handler of the custom controller?
Also, this seems like a very temporary workaround to me. The very next thing that we will have to fix, will be the 'edit' permission. And this does not have a custom controller.

Why not implement something like node_strict_access of Og but for entities? then we will have global coverage for all entities.
In any case, global permission management should not be handled in the controller. It should be generic. Otherwise we will be hunting ghosts until the end of a project and we are going to be filled with workarounds like this.